lib/api/core/notifier.js

```diff
@@ -258,8 +258,42 @@ class NotifierController {
    * @param {Array} ids - list of deleted document IDs
    */
   notifyDocumentDelete(request, ids) {
-    return Bluebird.each(ids, (id, callback) => {
-      return this.kuzzle.services.list.internalCache.search(this.cacheKeyPrefix + id)
+    return Bluebird.all(ids.map(id => {
+      const
+        getRequest = new Request({
```

Comment:
@ballinette @jenow , to me, the original aim of this issue was not about sending the body or not but to be able to notify clients on deletions at all.

Problem is: we are relying on the cache to detect the notifications to be sent. If a document has no entry in the cache, then no notification is sent.

Let's consider the following situation:

1. I am the only user to subscribe to one collection (no filter).
2. Another user deletes a document from this collection

In this situation, I won't get any notification at all.

@jenow : 
* We should, in case the document in cache, call, the DSL and the matching rooms in cache
* I don't see any reason why we would need to remove any entry from the cache

@ballinette , @jenow : I believe the question about if we should notify the whole document should be discussed with the team.
My personnal point of view if that, as it requires some extra work for Kuzzle, it should not be done by default, in which case, users should eventually be able to search the recycle bin in case they need to get the full document.